spec/controllers/market_lists_controller_spec.rb

```diff
@@ -0,0 +1,87 @@
+require 'rails_helper'
+
+RSpec.describe MarketListsController, type: :controller do
+  describe 'GET #index' do
+    it 'assigns @market_lists ordered by market_date' do
+      market_list1 = MarketList.create(name: 'Lista 1', market_date: Date.today)
+      market_list2 = MarketList.create(name: 'Lista 2', market_date: Date.yesterday)
+
+      get :index
+
+      expect(response).to be_successful
+      expect(assigns(:market_lists)).to eq([market_list1, market_list2])
+    end
+  end
+
+  describe 'POST #create' do
+    context 'with valid parameters' do
+      it 'creates a new market list and redirects to the index page' do
+        post :create, params: { market_list: { name: 'Nova Lista', market_date: Date.tomorrow } }
+
+        expect(response).to redirect_to(action: :index)
+        expect(flash[:success]).to be_present
+      end
+    end
+
+    context 'with invalid parameters' do
+      it 'does not create a new market list and render the new page' do
+        post :create, params: { market_list: { name: '', market_date: nil } }
+
+        expect(response).to render_template(:new)
+        expect(flash[:danger]).to be_present
+      end
+    end
+  end
+
+  describe 'PATCH #update' do
+    it 'updates market list and redirects to viewing page' do
```

Comment:
I believe it should have different context here (like you did on POST)

    context 'with valid parameters' do
     context 'with invalid parameters' do